src/library/client.lua

```diff
@@ -8,11 +8,76 @@ operating system.
 
 One of the main uses of using client details is to check its capabilities. As such,
 the bulk of this library is helper functions to determine what the client supports.
-All functions to check a client's capabilities should start with `client.supports_`.
-These functions don't accept any arguments, and should always return a boolean.
 ]] --
 local client = {}
 
+local function to_human_string(feature)
+    return string.gsub(feature, "_", " ")
+end
+
+local function requires_later_plugin_version(feature)
+    if feature then
+        return "This script uses " .. to_human_string(feature) .. "which is only available in a later version of RGP Lua. Please update RGP Lua instead to use this script."
+    end
+    return "This script requires a later version of RGP Lua. Please update RGP Lua instead to use this script."
+end
+
+local function requires_rgp_lua(feature)
+    if feature then
+        return "This script uses " .. to_human_string(feature) .. " which is not available on JW Lua. Please use RGP Lua instead to use this script."
+    end
+    return "This script requires RGP Lua, the successor of JW Lua. Please use RGP Lua instead to use this script."
+end
+
+local function requires_plugin_version(version, feature)
+    if version <= 0.54 then
+        if feature then
+            return "This script uses " .. to_human_string(feature) .. " which requires RGP Lua or JW Lua version " .. version ..
+                       " or later. Please update your plugin to use this script."
+        end
+        return "This script requires RGP Lua or JW Lua version " .. version .. " or later. Please update your plugin to use this script."
+    end
+    if feature then
+        return "This script uses " .. to_human_string(feature) .. " which requires RGP Lua version " .. version .. " or later. Please update your plugin to use this script."
+    end
+    return "This script requires RGP Lua version " .. version .. " or later. Please update your plugin to use this script."
+end
+
+local function requires_finale_version(version, feature)
+    return "This script uses " .. to_human_string(feature) .. ", which is only available on Finale " .. version .. " or later"
+end
+
+local features = {
+    clef_change = {
+        test = finenv.StringVersion >= "0.60",
+        error = requires_plugin_version("0.58", "a clef change"),
+    },
+    custom_key_signature = {
+        test = finenv.IsRGPLua and finale.FCKeySignature.__class.CalcTotalChromaticSteps,
+        error = requires_later_plugin_version("a custom key signature"),
+    },
+    ["FCCategory::SaveWithNewType"] = {
```

Comment:
I like this. Can the others be changed too where applicable? Eg `modeless_dialog` > `FCCustomLuaWindow::ShowModeless`, `retained_state` > `finenv.RetainLuaState`, etc, etc

It'll be more intuitive, where possible, rather than having to remember/look up what the feature has been called.

Reply:
I agree.